### PR TITLE
Updating istio/istio dependency

### PR DESCRIPTION
**Pull request by nmittler**

### CODE REVIEW

File: bin/init.sh [hunk 1]
```diff
@@ -21,6 +21,9 @@ set -o pipefail
 export ISTIO_REMOTE=${ISTIO_REMOTE:-origin}
 export ISTIO_BRANCH=${ISTIO_BRANCH:-master}
 
+echo "Updating the istio/istio dependency to the latest ${ISTIO_BRANCH} branch ..."
+go get istio.io/istio@${ISTIO_BRANCH}
```
Comment (howardjohn) [inline bin/init.sh]: I don't think we want to do this automatically? Since then what is checked in and what is tested are two different things?

Comment (ericvn) [reply bin/init.sh]: I suspect that we will need to add something in the branching instructions, but I would think that currently we would want to be testing against the `release-1.5` branch (set when the release branch is created for istio/istio). Once we created the 1.5 doc branch we would revert to testing against `master` (release-1.5 istio.io branch continues to get against 1.5).

**Comment by nmittler** [comment]
@howardjohn @ericvn PTAL

The script now downloads the same commit of Istio used in the go dependency.

File: Makefile.core.mk [hunk 2]
```diff
@@ -26,15 +24,26 @@ GO ?= go
 GOARCH_LOCAL := $(TARGET_ARCH)
 GOOS_LOCAL := $(TARGET_OS)
 
+# ISTIO_IMAGE_VERSION stores the prefix used by default for the Docker images for Istio.
+# For example, a value of 1.6-dev will assume a default TAG value of 1.6-dev.<SHA>
+ISTIO_IMAGE_VERSION ?= 1.6-dev
```
Comment (howardjohn) [inline Makefile.core.mk]: I think its 1.6-alpha

Comment (howardjohn) [reply Makefile.core.mk]: for example: gcr.io/istio-testing/pilot:1.6-alpha.a11761a194325cc1af23e9ab53110a0369e69fde

File: Makefile.core.mk [hunk 3]
```diff
@@ -125,12 +132,15 @@ foo2:
...
 .PHONY: init
 init:
+	@echo "ISTIO_SHA = ${ISTIO_SHA}"
+	@echo "HUB = ${HUB}"
+	@echo "TAG = ${TAG}"
 	bin/init.sh
 
 include tests/tests.mk
 
 # make lint-yaml seems to fail with pipefail, so remove now.
```
Comment (sdake) [inline Makefile.core.mk]: please file a bug and assign to me. Not super satisfied with the workaround in: https://github.com/istio/common-files/pull/200

Comment (nmittler) [reply Makefile.core.mk]: I don't seem able to create issues on https://github.com/istio/common-files/ ... permissions thing?

**Comment by howardjohn** [comment]
you need to do it in the main repo

File: Makefile.core.mk (hunk 3 above)
Comment (nmittler) [reply Makefile.core.mk]: Sure thing, @sdake  ... https://github.com/istio/istio/issues/21854

**Comment by nmittler** [comment]
@howardjohn ah thanks :)

**Pull request merged**